integration/spark/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchemaCommon.scala

```diff
@@ -275,6 +275,64 @@ class AlterTableColumnSchemaGenerator(
         allColumns ++= Seq(columnSchema)
       }
       newCols ++= Seq(columnSchema)
+      if (DataTypes.isArrayType(columnSchema.getDataType)) {
+        columnSchema.setNumberOfChild(field.children.size)
+        val childField = field.children.get(0)
+        // TODO: should support adding multi-level complex columns: CARBONDATA-4164
+        if (childField.children != Some(null)) {
```

Comment:
```suggestion
        if (!childField.children.contains(null)) {
```
Handle in all places

Reply:
done